packages/back-end/src/integrations/SqlIntegration.ts

```diff
@@ -1355,25 +1298,70 @@ export default abstract class SqlIntegration
     return `LEAST(${cap}, ${value})`;
   }
 
-  private getAggregateMetricColumn(metric: MetricInterface, alias = "m") {
+  private addPrePostTimeFilter(
+    col: string,
+    timePeriod: MetricAggregationType
+  ): string {
+    const mcol = `m.timestamp`;
+    if (timePeriod === "pre") {
+      return `${this.ifElse(`${mcol} < d.preexposure_end`, `${col}`, `NULL`)}`;
+    }
+    if (timePeriod === "post") {
+      return `${this.ifElse(
+        `${mcol} >= d.conversion_start`,
+        `${col}`,
+        `NULL`
+      )}`;
+    }
+    return `${col}`;
+  }
+
+  private getAggregateMetricColumn(
+    metric: MetricInterface,
+    metricTimeWindow: MetricAggregationType = "post"
+  ) {
     if (metric.type === "binomial") {
-      return "1";
+      return `MAX(${this.addPrePostTimeFilter("1", metricTimeWindow)})`;
     }
 
     const queryFormat = this.getMetricQueryFormat(metric);
     if (queryFormat === "sql") {
-      return this.capValue(
-        metric.cap,
-        metric.aggregation || `SUM(${alias}.value)`
-      );
+      let aggregation = `SUM(${this.addPrePostTimeFilter(
+        `m.value`,
+        metricTimeWindow
+      )})`;
+
+      if (metric.aggregation) {
+        if (Number(metric.aggregation)) {
+          return `MAX(${this.addPrePostTimeFilter(
+            `${aggregation}`,
```

Comment:
Should this be `metric.aggregation` instead of `aggregation`?

Reply:
To that end, this is one of the things I'd want to unit test. I guess this method is getting gnarly enough that setting up unit testing for it makes sense. I'll see about adding that today.

Reply:
Yes.  And the fact that `this.addPrePostTimeFilter` and `this.capValue` are called so many times within this function makes me think the abstraction level isn't quite right yet.